File: android/src/main/kotlin/cz/appsdevteam/gptom/models/PluginResponse.kt
package cz.appsdevteam.gptom.models

import cz.appsdevteam.gptom.core.JsonKeys
import cz.appsdevteam.gptom.models.ResultCodes

sealed class PluginResponse {
    data class Success(val data: Any?) : PluginResponse()

    data class Error(val code: String, val message: String, val data: Any? = null) : PluginResponse() {
        fun toEventData(): Map<String, Any?> = mapOf(
            JsonKeys.code to code,
            JsonKeys.message to message,
            JsonKeys.data to data,
        )
    }

    fun toMap(): Map<String, Any?> = when (this) {
        is Success -> mapOf(JsonKeys.code to ResultCodes.OK, JsonKeys.data to data)
        is Error -> mapOf(JsonKeys.code to code, JsonKeys.message to message, JsonKeys.data to data)
    }
}

File: ios/Classes/models/PluginResponse.swift
enum PluginResponse {
    case success(_ data: Any?)
    case error(code: String, message: String)

    func toMap() -> [String: Any] {
        switch self {
        case .success(let data):
            return [
                JsonKeys.code: ResultCodes.ok,
                JsonKeys.data: data ?? NSNull(),
            ]
        case .error(let code, let message):
            return [
                JsonKeys.code: code,
                JsonKeys.message: message,
                JsonKeys.data: NSNull(),
            ]
        }
    }
}
